plugin/proxy/grpc_test.go

```diff
@@ -59,6 +64,97 @@ func TestStartupShutdown(t *testing.T) {
 	}
 }
 
+func TestRunAQuery(t *testing.T) {
+	grpclog.SetLogger(discard{})
+
+	upstream := &staticUpstream{
+		from: ".",
+		HealthCheck: healthcheck.HealthCheck{
+			Hosts: pool(),
+		},
+	}
+	g := newGrpcClient(nil, upstream)
+	upstream.ex = g
+
+	p := &Proxy{}
+	p.Upstreams = &[]Upstream{upstream}
+
+	err := g.OnStartup(p)
+	if err != nil {
+		t.Errorf("Error starting grpc client exchanger: %s", err)
+		return
+	}
+	// verify the client is usable, or an error is properly raised
+	state := request.Request{W: &test.ResponseWriter{}, Req: new(dns.Msg)}
+	g.Exchange(context.TODO(), "localhost:10053", state)
+
+	// verify that you have proper error if the hostname is unknwn or not registered
+	_, err = g.Exchange(context.TODO(), "invalid:10055", state)
+	if err == nil {
+		t.Errorf("Expecting a proper error when querying gRPC client with invalid hostname : %s", err)
+	}
+
+	err = g.OnShutdown(p)
+	if err != nil {
+		t.Errorf("Error stopping grpc client exchanger: %s", err)
+		return
+	}
+}
+
+func TestRunAQueryOnSecureLinkWithInvalidCert(t *testing.T) {
+	grpclog.SetLogger(discard{})
+
+	upstreamHostname := "localhost:43001"
+	upstream := &staticUpstream{
+		from: ".",
+		HealthCheck: healthcheck.HealthCheck{
+			Hosts: []*healthcheck.UpstreamHost{
+				{
+					Name: upstreamHostname,
+				}},
+		},
+	}
+
+	filename, rmFunc, err := test.TempFile("", validCert)
+	if err != nil {
+		t.Errorf("Error saving file : %s", err)
+		return
+	}
+	defer rmFunc()
+
+	tls, err := tls.NewTLSClientConfig(filename)
```

Comment:
just ignore the error here, as you control the cert's data

Reply:
ok